manifests/istio-telemetry/kiali-operator/templates/deployment.yaml

```diff
@@ -0,0 +1,87 @@
+{{- if .Values.kialiOperator.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kiali-operator
+  namespace: {{ .Values.kialiOperator.namespace }}
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.kialiOperator.replicaCount }}
+  selector:
+    matchLabels:
+      app: kiali-operator
+  template:
+    metadata:
+      name: kiali-operator
+      labels:
+        app: kiali-operator
+        release: {{ .Release.Name }}
+      annotations:
+        sidecar.istio.io/inject: "false"
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "8383"
+        {{- if .Values.kialiOperator.podAnnotations }}
+{{ toYaml .Values.kialiOperator.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: kiali-operator-service-account
+{{- if .Values.global.priorityClassName }}
+      priorityClassName: "{{ .Values.global.priorityClassName }}"
+{{- end }}
+{{- if .Values.kialiOperator.tolerations }}
+      tolerations:
```

Comment:
is this required? I think users can just use the k8s overlay to set advanced settings like this and we avoid the complexity. Same for most optional fields here like annotations, etc

@richardwxn can you confirm?

Reply:
I was following the examples of how things were previously done. If there are other ways to define tolerations and resources and the like outside of defining it directly in the deployment yaml here, we can do that instead.